apis/identity/v1alpha2/iamrole_referencers.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha2
+
+import (
+	"context"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IAMRoleARNReferencer is used to get the ARN from a referenced IAMRole object
+type IAMRoleARNReferencer struct {
+	corev1.LocalObjectReference `json:",inline"`
+}
+
+// GetStatus implements GetStatus method of AttributeReferencer interface
+func (v *IAMRoleARNReferencer) GetStatus(ctx context.Context, res resource.CanReference, reader client.Reader) ([]resource.ReferenceStatus, error) {
+	return getRoleStatus(ctx, v.Name, res, reader)
+}
+
+// Build retrieves and builds the IAMRoleARN
+func (v *IAMRoleARNReferencer) Build(ctx context.Context, res resource.CanReference, reader client.Reader) (string, error) {
+	role := IAMRole{}
+	nn := types.NamespacedName{Name: v.Name, Namespace: res.GetNamespace()}
```

Comment:
I actually don't think we'll _need_ any updates to this implementation once my simple resource class selection code lands - `GetNamespace()` will return empty string and be ignored by `reader.Get` because `IAMRole` will now be cluster scoped. We''ll probably want to remove the `Namespace` bits at some point but they won't break anything.